src/SFML/Graphics/View.cpp

```diff
@@ -81,7 +84,10 @@ void View::setCenter(const Vector2f& center)
 ////////////////////////////////////////////////////////////
 void View::setSize(const Vector2f& size)
 {
-    m_size = size;
+    FloatRect newRect(FloatRect(m_center - size / 2.f, size).getAbsoluteRect());
```

Comment:
Note that negative-sized views are a valid use case to flip vertically or horizontally the displayed portion of the world.